File: languages/C++/src/Person.cpp
#include "Person.hpp"
#include<cstring>

Person::Person(const char* name,int age)
{
    this->name = new char[strlen(name)+1];
    strcpy(this->name,name);
    this->age=age;
}

char* Person::getName()
{
    return name;
}

int Person::getAge()
{
    return age;
}

File: languages/TypeScript/lib/Person.ts
import Girl from "./Girl";
import { Xp } from "./Xp";

/**
 * 人
 */
export default class Person {
  /**
   * 姓名
   */
  name: string = "";
  /**
   * 年龄
   */
  age: number = 0;

  /**
   * 标签
   */
  tags: string[] = [];

  /**
   * XP 系统
   */
  xp = new Xp();

  /**
   * 说话
   * @param word
   */
  say(word: string) {
    console.log(word);
  }

  like(girl: Girl) {
    return this.xp.match(girl.tags);
  }
}
